### ISSUE TITLE
Does cargo web deploy respect CARGO_TARGET_DIR?

### ISSUE THREAD

**Issue by Boscop** [koute/cargo-web#157]
Does `cargo web deploy` respect `CARGO_TARGET_DIR`? 
IOW, will it deploy to `$CARGO_TARGET_DIR/deploy`? It doesn't seem like it..
**How can I deploy to a custom dir then?**

I want to fork [this buildpack](https://github.com/emk/heroku-buildpack-rust) so that it also builds my frontend with `cargo web deploy` but so that the frontend build will also benefit from the build cache. The buildpack's `compile` script sets `CARGO_TARGET_DIR` so that it's in the cache dir:
https://github.com/emk/heroku-buildpack-rust/blob/117a46b773279d96bb3dcc2fb21bcf8d2d194f9c/bin/compile#L124

**Comment by koute** [comment]
Yes. It should respect `$CARGO_TARGET_DIR`, but it will copy the files to `$CARGO_TARGET_DIR/deploy`.

There is no way of specifying a custom deployment directory right now, but you can fake it with a symlink if you need it. Just symlink `$CARGO_TARGET_DIR/deploy` to wherever you want it deployed.

**Comment by Boscop** [comment]
@koute I'm deploying to heroku and I want to embed my frontend files with [rust-embed](https://github.com/pyros2097/rust-embed) but for that I have to specify a fixed dir in the proc-macro attribute because env vars aren't allowed in attributes. 
As long as `$CARGO_TARGET_DIR` is `target`, it works:
```rust
#[derive(RustEmbed)]
#[folder = "target/deploy/"] // not possible to use $CARGO_TARGET_DIR/deploy here
struct Assets;
```
But it wouldn't work on heroku with the build cache that uses a different `$CARGO_TARGET_DIR`. It would only work if proc-macro attributes allowed env vars:
```rust
#[folder = concat!(env!("CARGO_TARGET_DIR"), "/deploy/")]
```
But that's not possible.

So can you **please** add support for outputting to a custom directory? :)

**Comment by koute** [comment]
FYI. I've added `-o/--output` option to the `deploy` subcommand on `master`.

**Comment by koute** [comment]
Released as 0.6.22
